src/main/rx/crsf.c

```diff
@@ -138,36 +167,45 @@ STATIC_UNIT_TESTED void crsfDataReceive(uint16_t c)
 
     if (crsfFramePosition < fullFrameLength) {
         crsfFrame.bytes[crsfFramePosition++] = (uint8_t)c;
-        crsfFrameDone = crsfFramePosition < fullFrameLength ? false : true;
-        if (crsfFrameDone) {
-            crsfFramePosition = 0;
+        if (crsfFramePosition == fullFrameLength) {
+            crsfFramePosition = 0;     
+            if (!crsfConsumerReady) {
+                resetRingBuffer();
+            } else {
+                rbufWriteData(crsfRingBuffer.ring, crsfFrame.bytes, fullFrameLength);
+                crsfRingBuffer.frames++;
+            }
         }
     }
 }
 
-STATIC_UNIT_TESTED uint8_t crsfFrameCRC(void)
+STATIC_UNIT_TESTED uint8_t crsfFrameStatus(void)
 {
-    // CRC includes type and payload
-    uint8_t crc = crc8_dvb_s2(0, crsfFrame.frame.type);
-    for (int ii = 0; ii < crsfFrame.frame.frameLength - CRSF_FRAME_LENGTH_TYPE_CRC; ++ii) {
-        crc = crc8_dvb_s2(crc, crsfFrame.frame.payload[ii]);
+    crsfFrame_t frame;
+
+    if (!crsfConsumerReady) {
```

Comment:
The reason for this is that when the FC is powered on with the transmitter on, the RX dumped upwards of 10 frames into the buffer before the status function fired off.  This was a way of informing the ISR function to start buffering instead of overflowing the ring.

Reply:
which brings up another question -- The RX is sending these frames in a way that the FC cannot syncronize.  While I can't see it in a debugger (only happens on F4s and I don't have a means of debugging a F4), I can surmise by the behavior of the FC that the frames are being built in the middle of the frame instead of at the beginning.  Any thoughts on this?

Reply:
There is no check for frame length (except for implicit 8bit limit + CRSF_FRAME_LENGTH_ADDRESS + CRSF_FRAME_LENGTH_FRAMELENGTH). Maybe something gets overwritten.

After first error frames will be received randomly until interframe gap provides resync.

Reply:
any way of shifting the buffer to find this resync?

Reply:
or should it be sanity checked for length < max length and let the crc dump it later?

Reply:
Not sure ... checking crc and bailing out on error (wait or sync state) is one possibility ..
Even with packet error only frames matching length should be committed into ringbuffer, others are silently discarded when interframe gap arrives.

ringbuff code should check for for overflow - it may be main problem now ...

Reply:
Originally I wrote the rbuf writes to return bool.. false when there isn't room in the buffer.  This works if you're only committing one thing at a time.  In this case, it may be wise to prepend the length for easier consumption later.  This would require two writes.  Maybe a simple check of rbufWriteBytesRemaining prior to committing anything to the buffer would help.  If there's room, add it.  Let the CRC determine whether or not it passes the sanity check.

Reply:
Explicit check for available space should do (discard if frame won't fit) ... but as I said before, ringbuff is overkill.